### DISCUSSION TITLE
Different results between OMEdit and FMU?

### DISCUSSION THREAD

**Discussion by Amuri44** [ThummeTo/FMI.jl#89] [Q&A]
I have a model `R` in in OpenModelica, in which it is a user-defined block.

**Comment by ThummeTo** [comment]
Dear @Amuri44 ,

what I see looks fine so far, but the code is not complete, e.g. is `dt=1e-6`?
Further information on the FMI2-calling convention ("state-machine") can be found in the [FMI2-specification](https://fmi-standard.org/).

Further you could check via `fmiGetReal(comFMU, ["aV.Pm","aV.Vs","aV.Vic"])` if all parameters are set correctly after `fmiExitInitializationMode(comFMU)`. How does the output differ? Could you provide a plot and/or further information? Do you log `Pmm, Vff` inside the loop? It is hard to debug without further information. 

Please note, that this could be done (more efficient) with `fmiSimulateCS` as well.

Regards!

**Reply by Amuri44** [reply]
- Yes, `dt=1e-6` (I have added it in the previous code)

**Reply by ThummeTo** [reply]
The provided code overwrites the variables `Pmm, Vff` in every simulation step. This leads to a flat curve.

What is the simulation result if you use 
```
success, recordedValues = fmi2SimulateCS(comFMU, t_start, t_stop;
                        tolerance=mySimulationTolerance,
                        dt=1e-6,
                        recordValues=["aV.Pm","aV.Vf"],
                        setup=false,
                        reset=false,
                        inputValues=["aV.vd","aV.vq"],
                        inputFunction=myInputFunction)
```

with 

```
function myInputFunction(t)
   return [5.0, 5.0]
end
``` 

instead of your custom loop? The results are saved in `recordedValues`.
                        
Finally if you need a high simulation resolution you could try changing the solver tolerance via `mySimulationTolerance`.

**Reply by ThummeTo** [reply]
`return [1.0,5.0, 5.0,1.0]`

**Reply by ThummeTo** [reply]
> > The provided code overwrites the variables `Pmm, Vff` in every simulation step. This leads to a flat curve.
> 
> Ok, how to prevent the overwrites?

Don't overwrite it ;-)
For example pushing the current value into an array at every simulation step.

**Reply by ThummeTo** [reply]
Could you check another simulation tool, like e.g. [fmpy](https://github.com/CATIA-Systems/FMPy)? Another fast check would be to import the FMU again into OMEdit and check if the FMU simulates like the original system.

It is hard to say where the error is coming from. The first question is if this error is caused by the FMU (exported by OMEdit) or by the simulation tool (FMI.jl).
